File: 프로그래머스/unrated/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.py
def solution(today, terms, privacies):
    today = list(map(int, today.split('.')))
    today = today[0] * 12 * 28 + today[1] * 28 + today[2]
    answer = []
    term = {}
    for te in terms:
        t, n = te.split()
        term[t] = int(n)
    
    idx = 1
    for privacie in privacies:
        date, t = privacie.split()
        date = list(map(int, date.split('.')))
        date = date[0] * 12 * 28 + date[1] * 28 + date[2] + term[t] * 28
        if today >= date:
            answer.append(idx)
        idx += 1
    return answer

File: 프로그래머스/1/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.cpp
#include <string>
#include <vector>
#include <sstream>
#include <iostream>
#include <unordered_map>

using namespace std;

vector<string> split(const string &str, char delimeter) {
    vector<string> result;
    stringstream ss(str);
    string token;
    while (getline(ss, token, delimeter)) result.push_back(token);
    return result;
}

int transformDate(const string & Date) {
    vector<string> parseDate = split(Date, '.');
    return stoi(parseDate[0]) * 12 * 28 + stoi(parseDate[1]) * 28 + stoi(parseDate[2]);
}

vector<int> solution(string today, vector<string> terms, vector<string> privacies) {
    vector<int> answer;
    unordered_map<string, int> termsMap;
    int todayNum = transformDate(today);
    for (const auto &term : terms) {
        vector<string> parsedTerm = split(term, ' ');
        termsMap[parsedTerm[0]] = stoi(parsedTerm[1]) * 28;
    }
    
    for (int i = 0; i < privacies.size(); i++) {
        vector<string> privacy = split(privacies[i], ' ');
        int dateNum = transformDate(privacy[0]) + termsMap[privacy[1]];
        if (todayNum >= dateNum) answer.push_back(i + 1);
    }
    
    return answer;
}